tests/middlewareTypes.test.tsx

```diff
@@ -112,7 +112,7 @@ describe('counter state spec (single middleware)', () => {
     }
     TestComponent
 
-    const _testSubtyping: StoreApi<object> = createStore(
+    const _testSubtyping: StoreApi<{ count: number }> = createStore(
```

Comment:
I am not sure if this is the right way to fix this test... The problem is that the `createStore` call creates a `StoreApi<{ count: number }>`, which used to be convertible to `StoreApi<object>` but no longer is (since `.set({}, true)` is disallowed)

Reply:
Me neither...

Let's see if @devanshj can respond.

Reply:
This test makes sure any store created is a subtype of `Store<object>` because that's how you'd write a function that takes any store...

```ts
const somePlugin = <S extends Store<object>>(s: S): S => {
   // ...
   return s;
}
somePlugin(someStore) // breaks after this change
```

Reply:
Does reverting it cause the test fail?
```suggestion
    const _testSubtyping: StoreApi<object> = createStore(
```